app/app.rb

```diff
@@ -0,0 +1,115 @@
+ENV['RACK_ENV'] ||= 'development'
+
+require 'sinatra/base'
+require 'sinatra/flash'
+require 'sinatra/partial'
+require_relative 'models/data_mapper_setup'
+require 'byebug'
+
+class Chitter < Sinatra::Base
+  register Sinatra::Flash
+  register Sinatra::Partial
+
+  use Rack::MethodOverride
+
+  enable :sessions
+  enable :partial_underscores
+
+  set :session_secret, "secret message"
+  set :partial_template_engine, :erb
+
+  get '/' do
+    @peeps = Peep.all
+    erb :'/peeps/index'
+  end
+
+  get '/users/new' do
+    @user = User.new
+    erb :'/users/new'
+  end
+
+  post '/users' do
+    @user = User.create(name: params[:name],
+                      username: params[:username],
+                      email: params[:email],
+                      password: params[:password],
+                      password_confirmation: params[:password_confirm])
+    if @user.id
+      session[:user_id]=@user.id
+      redirect '/'
+    else
+      flash.now[:error]=@user.errors.full_messages
+      erb :'/users/new'
+    end
+  end
+
+  get '/sessions/new' do
+    erb :'sessions/new'
+  end
+
+  post '/sessions' do
+    user = User.authenticate(params[:email], params[:password])
+    if user
+      session[:user_id]=user.id
+      flash.next[:success]='You are now logged in.'
+      redirect '/'
+    else
+      flash.now[:notice]='The email or password was incorrect'
+      erb :'/sessions/new'
+    end
+  end
+
+  delete '/sessions' do
+    session[:user_id] = nil
+    flash.keep[:notice]='You are now signed out'
+    redirect to '/'
+  end
+
+  post '/peeps' do
+    peep = Peep.create(message: params[:peep], user: current_user)
+    current_user.peeps << peep
+    current_user.save
+    redirect to '/'
+  end
+
+  get '/users/recover' do
+    erb :'/users/recover'
+  end
+
+  post '/users/recover' do
+    user = User.first(email: params[:email])
+    if user
```

Comment:
Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`. (https://github.com/bbatsov/ruby-style-guide#if-as-a-modifier)